pkg/gds/backup_test.go

```diff
@@ -0,0 +1,83 @@
+package gds_test
+
+import (
+	"io/ioutil"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/trisacrypto/directory/pkg/gds"
+	"github.com/trisacrypto/directory/pkg/gds/config"
+)
+
+// Test that the backup manager does not create backups if disabled.
+func (s *gdsTestSuite) TestBackupManagerDisabled() {
+	conf := gds.MockConfig()
+	conf.Backup = config.BackupConfig{
+		Enabled:  false,
+		Interval: time.Millisecond,
+		Storage:  "testdata/backup",
+		Keep:     1,
+	}
+	s.SetConfig(conf)
+	defer s.ResetConfig()
+	s.LoadEmptyFixtures()
+	require := s.Require()
+
+	// Start the backup manager
+	var wg sync.WaitGroup
+	wg.Add(1)
+	go s.svc.BackupManager(nil, &wg)
+
+	// Wait for the backup interval to elapse
+	time.Sleep(s.svc.GetConf().Backup.Interval * 2)
+
+	// Backup should not be created
+	backupDir := s.svc.GetConf().Backup.Storage
+	require.NoDirExists(backupDir)
+
+	// Make sure the backup manager is stopped before we exit
+	wg.Wait()
```

Comment:
Because our expectation is that the backup routine is not running, this could replace the Sleep on Line 33. I do think we should have the Wait before we check the backupDir if you want to keep the sleep, just to prevent a race condition.